sdk/src/Core/Amazon.Auth/AccessControlPolicy/Internal/JsonPolicyReader.cs

```diff
@@ -97,24 +97,19 @@ private static void convertPrincipals(Statement statement, JsonElement jStatemen
                 {
                     if (jPrincipal.Value.ValueKind == JsonValueKind.String)
                     {
-                        if (jPrincipal.Value.GetString().Equals("*"))
-                        {
-                            statement.Principals.Add(Principal.Anonymous);
-                        }
-                        else
-                        {
-                            // Don't strip '-' and assume the policy being deserialized is already valid.
-                            Principal principal = new Principal(jPrincipal.Name, jPrincipal.Value.GetString());
-                        }
+                        // Don't strip '-' and assume the policy being deserialized is already valid.
+                        Principal principal = new Principal(jPrincipal.Name, jPrincipal.Value.GetString());
+                        statement.Principals.Add(principal);
                     }
                     else if (jPrincipal.Value.ValueKind == JsonValueKind.Array)
                     {
-                        foreach (JsonElement arrayElement in jPrincipals.EnumerateArray())
+                        foreach (JsonElement arrayElement in jPrincipal.Value.EnumerateArray())
```

Comment:
In addition to adding the tests, I also had to change this in order for them to pass.

Running a build through our internal system now.

Reply:
@dscpinheiro haha I just pushed the same 👯 I will undo my changes